kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -53,8 +55,30 @@ class SparkProcessBuilder(
     Paths.get(sparkHome, "bin", SPARK_SUBMIT_FILE).toFile.getCanonicalPath
   }
 
+  initialize
+
   override def mainClass: String = "org.apache.kyuubi.engine.spark.SparkSQLEngine"
 
+  /**
+   * Add or override spark.master if KUBERNETES_SERVICE_HOST and
+   * KUBERNETES_SERVICE_PORT are defined
+   */
+  protected def initialize(): Unit = {
+    try {
+      if (conf.get(KUBERNETES_MASTER_OVERRIDE)) {
+        val kubernetesServiceHost = sys.env.get(KUBERNETES_SERVICE_HOST)
+        val kubernetesServicePort = sys.env.get(KUBERNETES_SERVICE_PORT)
+        if (kubernetesServiceHost.isDefined && kubernetesServicePort.isDefined) {
+          val masterURL = s"k8s://https://${kubernetesServiceHost.get}:${kubernetesServicePort.get}"
```

Comment:
w/ hardcoded "https", how to support "http"?

Reply:
After checking "https://kubernetes.io/docs/concepts/architecture/control-plane-node-communication/#node-to-control-plane", the API server is configured to listen for remote connections on a secure HTTPS port (typically 443), so just set https here.

Reply:
If user want to use http, he could set `spark.master` explicitly, which we will respect and will not be overridden after changing of my code.